presto-main/src/main/java/com/facebook/presto/sql/gen/RowExpressionCompiler.java

```diff
@@ -72,7 +73,12 @@ public RowExpressionCompiler(
 
     public BytecodeNode generate(RowExpression rowExpression, Scope scope)
     {
-        return rowExpression.accept(new Visitor(), new Context(scope));
+        return rowExpression.accept(new Visitor(), new Context(scope, Optional.empty()));
+    }
+
+    public BytecodeNode generate(RowExpression rowExpression, Scope scope, Class lambdaInterface)
```

Comment:
Make the last argument `Optional<Class>`. Have the other `generate` method delegate to this one.

In general, we avoid having multiple overloads of a method. However, there is no black and white rules. When we choose to have overloads, having all of them delegate to the same method helps to make sure that the method signatures are sane and extensible. Consider what happens if you need to add another Optional argument here.